Document build acceleration log analysis (#8802)
dotnet/project-system: docs/build-acceleration.md

## Debugging

### Enable logging

Build acceleration runs with the FUTDC, and outputs details of its operation in the build log. To enable this logging:

> > Tools | Options | Projects and Solutions | .NET Core

Reviewer: I think you mean:

```suggestion
> Tools | Options | Projects and Solutions | SDK-Style Projects
```

Reviewer: Also in the caption of the image below this line, but I couldn't leave a comment on that. 🤷🏻‍♂️

Author: Good catch. This was copied from the FUTDC document, so I'll update it there too.

- `Minimal` produces a single message per out-of-date project.
- `Info` and `Verbose` provide increasingly detailed information about the inner workings of the check, which are useful for debugging.

### Validate builds are accelerated

> If build acceleration cannot be enabled for any of the reasons given below, builds continue to work as before but will be slower.

Reviewer: I think you can just drop the "will be slower" part:

```suggestion
If build acceleration cannot be enabled for any of the reasons given below, builds continue to work as before.
```

They will be slower than if build acceleration could be enabled, but they won't be slower than they were before the build acceleration feature came along.

The following prerequisites exist for build acceleration:

- You are running Visual Studio 2022 version 17.5 or later.

> - The project is an SDK-style .NET project.

Reviewer 2: Have we tested this for VB/F#?

Author: VB: yes, because we are dogfooding it in this repo.
F#: not yet. I'll make sure that happens.

- All projects it references (directly and transitively) are also SDK-style .NET projects.

The following steps will validate that build acceleration is working correctly for a given project.

1. Ensure `Verbose` logging is enabled (see [Enable logging](#enable-logging)).
1. Build the project to make it up-to-date.
1. Modify source of a referenced project (either a direct reference or transitive reference).
1. Build the project again.

Looking through the build output with the following points in mind:

- ℹ️ If you see:

   > This project appears to be a candidate for build acceleration. To opt in, set the 'AccelerateBuildsInVisualStudio' MSBuild property to 'true'.

>    Then the project does not specify the `AccelerateBuildsInVisualStudio` property, or its value was not `true` or `false`, and the project like it would benefit from build acceleration. If the project cannot use build acceleration for any reason, this message can be suppressed by setting the property to `false` explicitly. See [configuration](#configuration) to learn how to configure build acceleration correctly.

Reviewer: Did you mean this?

```suggestion
   Then the project does not specify the `AccelerateBuildsInVisualStudio` property, or its value was not `true` or `false`, and the project likely would benefit from build acceleration. If the project cannot use build acceleration for any reason, this message can be suppressed by setting the property to `false` explicitly. See [configuration](#configuration) to learn how to configure build acceleration correctly.
```
